src/cursor/cur_version.c

```diff
@@ -307,15 +307,13 @@ __curversion_next_int(WT_CURSOR *cursor)
             WT_ERR(__wt_illegal_value(session, page->type));
         }
 
-        /* Get the ondisk value. */
-        ret = __wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &cbt->upd_value->tw
-#ifdef HAVE_DIAGNOSTIC
-          ,
-          NULL
-#endif
-        );
-        WT_ERR(ret);
-
+        /*
+         * Get the ondisk value. If we see an overflow removed value, return the overflow removed
+         * value for debugging.
+         */
+        WT_ERR_ERROR_OK(
+          __wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &cbt->upd_value->tw),
+          WT_RESTART, false);
```

Comment:
I changed the code here to still return the overflow removed value in version cursor. @agorrod